core/cmd/run_cmd.go

```diff
@@ -44,7 +44,7 @@ func runServices(ctx context.Context, cfg *config.Config, stream bool, xchain bo
 	var xchainService server.XChain
 	var wg sync.WaitGroup
 	if xchain {
-		xchainService, err = server.New(ctx, cmdConfig, nil, 1, metricsRegistry)
+		xchainService, err = server.New(ctx, cfg, nil, nil, 1, metricsRegistry)
```

Comment:
minor logical fix - cmdConfig is the global config, not what is passed into this method. The cfgs refer to the same object, so there is no change in behavior.